src/types/route-spec.ts

```diff
@@ -2,26 +2,67 @@ import type { z } from "zod"
 import type {
   AccumulateMiddlewareChainResultOptions,
   MiddlewareChain,
-  Middleware,
   MapMiddlewares,
 } from "./middleware"
 import type {
   GetAuthMiddlewaresFromGlobalSpec,
   GlobalSpec,
 } from "./global-spec"
-import type { EdgeSpecRouteFn } from "./web-handler"
-import type { InferRecordKey } from "./util"
+import type {
+  EdgeSpecCustomResponse,
+  EdgeSpecFormDataResponse,
+  EdgeSpecJsonResponse,
+  EdgeSpecRouteFn,
+  HTTPMethods,
+} from "./web-handler"
 
 export type RouteSpec<AuthMiddlewares extends string> = {
-  methods: readonly ("GET" | "POST" | "PUT" | "PATCH" | "DELETE")[]
+  methods: readonly HTTPMethods[]
 
   jsonBody?: z.ZodTypeAny
+  formData?: z.ZodObject<any>
+  queryParams?: z.ZodObject<any>
+  commonParams?: z.ZodObject<any>
+
   jsonResponse?: z.ZodTypeAny
+  formDataResponse?: z.ZodObject<any>
+  wwwFormUrlEncodedResponse?: z.ZodObject<any>
+  customResponseMap?: Record<string, z.ZodTypeAny>
 
   auth: AuthMiddlewares | readonly AuthMiddlewares[] | "none"
   middlewares?: MiddlewareChain
+
+  /**
+   * add x-fern-sdk-return-value to the openapi spec, useful when you want to return only a subset of the response
+   */
+  sdkReturnValue?: string | string[]
```

Comment:
🐧 I'd probably omit this for now, would be nice to have a more generic method for transforming parts of the generated spec